### PR TITLE
fix: Use display swap on Google Fonts

### LINKED ISSUE #13563: Default Google Font API optimization required by Spartacus OOTB!
The default GAPI Font used by Spartacus to need to have a property `&display=swap` at the end to resolve page speed score issue.

Page speed issue shows issue : 
Gfont_spartacus OOTB issue

As per:  https://web.dev/font-display/?utm_source=lighthouse&utm_medium=unknown#google-fonts
Recommended Fix : <link href="https://fonts.googleapis.com/css?family=Roboto:400,700&display=swap" rel="stylesheet">

Issue at OOTB level : 
file : node_modules\@spartacus\styles\scss\_fonts.scss
line : @import url('https://fonts.googleapis.com/css?family=Open+Sans:300,400,600,700'); << it do not have swap option

Kindly let us know if this can be overridden easily to not to appear in page speed. 

Project reference hierarchy to track the issue >> src\styles.scss>>node_modules\@spartacus\styles\_index.scss>>node_modules\@spartacus\styles\scss\app.scss>>node_modules\@spartacus\styles\scss\_fonts.scss>>node_modules\@spartacus\styles\scss\theme\sparta\_fonts.scss

### PR DESCRIPTION
**Pull request by Zeyber**
Closes: #13563

### CODE REVIEW

File: projects/storefrontstyles/scss/theme/sparta/_fonts.scss [hunk 1]
```diff
@@ -1 +1 @@
-@import url('https://fonts.googleapis.com/css?family=Open+Sans:300,400,600,700');
+@import url('https://fonts.googleapis.com/css?family=Open+Sans:300,400,600,700&display=swap');
```
Comment (Matejk00) [inline projects/storefrontstyles/scss/theme/sparta/_fonts.scss]: I'm not sure if we can set it up like this. 

In my opinion as we want to have it released in minor proposed solution may be a breaking change by forcing existing customers after update to use `swap` strategy by default.

Instead, my proposal is maybe to expose new variable which default value will be `auto`, but `swap` can be set before Spartacus SCSS imports. Example: https://stackoverflow.com/a/59128372/13845333

The best option will be to ask Mikel (@morganm58) about it before.

**Review by Matejk00 — CHANGES_REQUESTED** [review]

File: projects/storefrontstyles/scss/theme/sparta/_fonts.scss [hunk 2]
```diff
@@ -1 +1,5 @@
-@import url('https://fonts.googleapis.com/css?family=Open+Sans:300,400,600,700');
+// TODO: Replace `$font-url` with swap url in 5.0 to use swap strategy by default.
+// $font-url: 'https://fonts.googleapis.com/css?family=Open+Sans:300,400,600,700&display=swap';
+$font-url: 'https://fonts.googleapis.com/css?family=Open+Sans:300,400,600,700';
+
+@import url('#{$fontUrl}');
```
Comment (Matejk00) [inline projects/storefrontstyles/scss/theme/sparta/_fonts.scss]:
```
ModuleBuildError: Module build failed (from ./node_modules/@angular-devkit/build-angular/node_modules/sass-loader/dist/cjs.js):
SassError: Undefined variable.
  ╷
5 │ @import url('#{$fontUrl}');
```

Seems that we are trying to import wrong variable. Shouldn't be `$font-url` instead?

**Review by Matejk00 — CHANGES_REQUESTED** [review]

File: projects/storefrontstyles/scss/theme/sparta/_fonts.scss [hunk 3]
```diff
@@ -1 +1,9 @@
-@import url('https://fonts.googleapis.com/css?family=Open+Sans:300,400,600,700');
+@include forVersion(4, 4.99) {
```
Comment (Matejk00) [inline projects/storefrontstyles/scss/theme/sparta/_fonts.scss]: It's `forVersion(4, 4.99)` for purpose? Can't we use `forVersion(4.1)`?

Comment (Zeyber) [reply projects/storefrontstyles/scss/theme/sparta/_fonts.scss]: `forVersion(4.1)` doesn't work like that unfortunately. Changing the versions to `5` or `5.0` still also include the `forVersion(4.1)` function even though comments in `versioning.scss` claim it shouldn't include it in the next major.

File: projects/storefrontstyles/scss/theme/sparta/_fonts.scss (hunk 1 above)
Comment (morganm58) [reply projects/storefrontstyles/scss/theme/sparta/_fonts.scss]: After looking into this. This is the new default when you copy the code from Google fonts now. `$font-url: 'https://fonts.googleapis.com/css?family=Open+Sans:300,400,600,700&display=swap';` looks correct to me.

File: projects/storefrontstyles/scss/theme/sparta/_fonts.scss (hunk 2 above)
Comment (Matejk00) [inline projects/storefrontstyles/scss/theme/sparta/_fonts.scss]:
```suggestion
@import url('#{$font-url}');
```

**Review by Matejk00 — CHANGES_REQUESTED** [review]

**Review by Matejk00 — APPROVED** [review]

**Pull request merged**